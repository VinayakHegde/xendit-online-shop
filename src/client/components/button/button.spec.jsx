import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import { Button } from ".";

afterAll(() => {
  jest.clearAllMocks();
});
describe.each([
  ['enabled', false],
  ['disabled', true]
])('Button - %s', (a, b) => {

  it('should render without crash', () => {
    const onClick = jest.fn();
    const { container } = render(<Button disabled={b} type="button" onClick={onClick} className="My__button" title="Click me" />)
    expect(container).toMatchSnapshot();
  });
  it('should fire click event', () => {
    const onClick = jest.fn();
    const { getByTitle } = render(<Button disabled={b} type="button" onClick={onClick} className="My__button" title="Click me" />)
    const button = getByTitle('Click me');
    act(() => {
      fireEvent.click(button);
    });
    if (!b) {
      expect(onClick).toHaveBeenCalled();
    } else {
      expect(onClick).not.toHaveBeenCalled();
    }
  });
});
