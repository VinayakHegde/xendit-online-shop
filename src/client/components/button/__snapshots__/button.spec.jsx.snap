// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - disabled should render without crash 1`] = `
<div>
  <button
    class="Button My__button"
    disabled=""
    title="Click me"
    type="button"
  >
    Click me
  </button>
</div>
`;

exports[`Button - enabled should render without crash 1`] = `
<div>
  <button
    class="Button My__button"
    title="Click me"
    type="button"
  >
    Click me
  </button>
</div>
`;
